bin/region_mover.rb

```diff
@@ -493,11 +493,22 @@ def getFilename(options, targetServer, port)
   return filename
 end
 
+def getServersRSGroup(rsgroup_admin, hostname, port)
+  rsgroup = rsgroup_admin.getRSGroupOfServer(Address.fromParts(hostname,
+                                             java.lang.Integer.parseInt(port)))
+  # If the rsgroup is nil, that means this server belongs to no rsgroup.
+  # It should be already offline.
+  # Here we directly log and exit.
+  return rsgroup unless rsgroup.nil?
+  $LOG.info('The server ' + hostname + 'belongs to no rsgroup. Exit regions moving.')
+  exit 0
```

Comment:
The code and syntax looks good to me
>Here, I see one minor issue: if we go with this change, still the unload region would not make a graceful exit:
Can we instead call rsgroupAdmin.getRSGroupOfServer outside of getSameRSGroupServers method and if it returns nil, we can log a message and then exit gracefully?

I do question this design decision, though. If a server has regions on it, but fails to find any target servers to unload to, then that seems like a reason to exit with an error code.

So then I would propose that we explicitly check for this case. It's okay to exit with 0 if the server has no rsgroup AND has no regions. This would obviously require additional refactoring of the code, since currently we don't check for regions until after we obtain the rsgroup of the server. And in the case mentioned in HBASE-26596, I don't know whether a call to `getOnlineRegions` would succeed.

This is only my opinion, so feel free to ignore if you feel differently 🙂

Reply:
> It's okay to exit with 0 if the server has no rsgroup AND has no regions.

> I don't know whether a call to `getOnlineRegions` would succeed.

Hmm, I agree with first statement and that second statement would also be true.
How about we check something like this?
```
rsgroup = rsgroupAdmin.getRSGroupOfServer
if rsgroup.nil?
  try
    regions = admin.getOnlineRegions(server)
  except
    LOG.warn(.....)
    exit 0
  if not regions.nil? and regions.size == 0:
    exit 0
  else
    LOG.warn('Server is not part of any RSGroup and still hosting some regions')
    exit 5
```